src/server/graphql/mutations/stripeSucceedPayment.js

```diff
@@ -0,0 +1,46 @@
+import {GraphQLBoolean, GraphQLID, GraphQLNonNull} from 'graphql';
+import stripe from 'server/billing/stripe';
+import getRethink from 'server/database/rethinkDriver';
+import {errorObj} from 'server/utils/utils';
+import {PAID} from 'universal/utils/constants';
+
+
+export default {
+  name: 'StripeSucceedPayment',
+  description: 'When stripe tells us an invoice payment was successful, update it in our DB',
+  type: GraphQLBoolean,
+  args: {
+    invoiceId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'The stripe invoice ID'
+    }
+  },
+  resolve: async (source, {invoiceId}, {serverSecret}) => {
+    const r = getRethink();
+    const now = new Date();
+
+    // AUTH
+    if (serverSecret !== process.env.AUTH0_CLIENT_SECRET) {
+      throw new Error('Don’t be rude.');
+    }
+
+    // VALIDATION
+    const {customer: customerId, subscription, paid} = await stripe.invoices.retrieve(invoiceId);
+    const {metadata: {orgId}} = await stripe.customers.retrieve(customerId);
+    const org = await r.table('Organization').get(orgId);
+    if (!org) {
+      throw errorObj({_error: `Payment cannot succeed. Org ${orgId} does not exist for invoice ${invoiceId}`});
+    }
+    const {creditCard, stripeSubscriptionId} = org;
+    if (!paid || stripeSubscriptionId !== subscription) {
+      throw new Error(`Possible nefarious activity. Bad invoiceId received: ${invoiceId}`);
+    }
+
+    // RESOLUTION
+    await r.table('Invoice').get(invoiceId).update({
+      creditCard,
+      paidAt: now,
+      status: PAID
+    });
+  }
```

Comment:
if it were `new GraphQLNonNull(GraphQLBoolean)` GraphQL would return an array of errors with the first one saying the type was null when it couldn't be. since things are optional unless stated otherwise, it'll be AOK